### ISSUE TITLE
App keeps refreshing/searching endlessly

### ISSUE THREAD

**Issue by question-eng** [ImranR98/Obtainium#2019]
**Describe the bug**
The "bios lifeline animation" at the top plays forever and I cannot add new apps to obtainium

**To Reproduce**
Always, disabling app or rebooting does not fix

**Screenshots and Logs**

**Please complete the following information:**
 - Device: Pixel 8
 - OS: Android 15
 - Obtainium Version: 1.1.33

**Comment by ImranR98** [comment]
Did you have VLC, WhatsApp, and/or Telegram in your apps list?

**Comment by ImranR98** [comment]
If you install the latest release [v1.1.34](https://github.com/ImranR98/Obtainium/releases/tag/v1.1.34), your problem will be solved (assuming it's an issue with VLC like I suspect).

However, your VLC entry (if any) will no longer work. You'll need to remove it and re-add the config from here: https://apps.obtainium.imranr.dev/?categories=video_player

Also, your Obtainium logs (accessible from the settings page) will show an error message about "JSON compat modifiers". Please paste that error message here. Hopefully it will help me find/fix what went wrong so that people who update from `v1.1.32` later on won't have to delete/re-add VLC.

**Comment by Adhjie** [comment]
I have VLC in my phone, this is specific just to v1.1.32 currently or more?
I'll check those 3 apps in my phone, though for WhatsApp, I'm not sure if I could update from the gplay version. there's also meta updater for facebook, instagram, whatsapp. not sure if it will conflict or not.
I also just realized bitwarden-fdroid version will be the github version if 'set gplay as source when shizuku is on'.
but currently, I'll just check vlc for now.
will report later when I got the time.

**Comment by NicholasFlamy** [comment]
> If you install the latest release [v1.1.34](https://github.com/ImranR98/Obtainium/releases/tag/v1.1.34), your problem will be solved (assuming it's an issue with VLC like I suspect).
> 
> However, your VLC entry (if any) will no longer work. You'll need to remove it and re-add the config from here: https://apps.obtainium.imranr.dev/?categories=video_player
> 
> Also, your Obtainium logs (accessible from the settings page) will show an error message about "JSON compat modifiers". Please paste that error message here. Hopefully it will help me find/fix what went wrong so that people who update from `v1.1.32` later on won't have to delete/re-add VLC.

It seems like whoever updates the APK link on the VLC website hasn't bothered to update it in 2 versions but otherwise it works.

**Comment by BRBsoup** [comment]
I had no problem with my VLC or Obtainium, which I added through this link: 
https://artifacts.videolan.org/vlc-android/nightly-arm64-v4/

My Obtainium config: [Add To Obtainium: VLC Android Nightly Arm64](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22org.videolan.vlc.debug%22%2C%22url%22%3A%22https%3A%2F%2Fartifacts.videolan.org%2Fvlc-android%2Fnightly-arm64-v4%2F%22%2C%22author%22%3A%22artifacts.videolan.org%22%2C%22name%22%3A%22VLC%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Afalse%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22APKLinkHash%5C%22%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%5C%22%2C%5C%22versionDetection%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Afalse%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Atrue%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Atrue%7D%22%2C%22overrideSource%22%3Anull%7D)

Last VLC update from log:

`2024-12-05 08:54:18.447: info: Installing "org.videolan.vlc.debug" version "4.0.0-preview - 3050740" versionCode "13050740" (from existing version "4.0.0-preview - 3050740" versionCode "13050740")`
^ That was while still on Obtainium version 1.1.32 though, and there have been no new APK's added since, so...

**Comment by NicholasFlamy** [comment]
> I had no problem which I've noticed with my VLC or Obtainium, which I added through this link: https://artifacts.videolan.org/vlc-android/nightly-arm64-v4/
> 
> My Obtainium config: [Add To Obtainium: VLC Android Nightly Arm64](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22org.videolan.vlc.debug%22%2C%22url%22%3A%22https%3A%2F%2Fartifacts.videolan.org%2Fvlc-android%2Fnightly-arm64-v4%2F%22%2C%22author%22%3A%22artifacts.videolan.org%22%2C%22name%22%3A%22VLC%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Afalse%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22APKLinkHash%5C%22%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%5C%22%2C%5C%22versionDetection%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Afalse%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Atrue%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Atrue%7D%22%2C%22overrideSource%22%3Anull%7D)
> 
> Last VLC update from log:
> 
> `2024-12-05 08:54:18.447: info: Installing "org.videolan.vlc.debug" version "4.0.0-preview - 3050740" versionCode "13050740" (from existing version "4.0.0-preview - 3050740" versionCode "13050740")`
> 
> ^ That was while still on Obtainium version 1.1.32 though, and there have been no new APK's added since, so...

THANK YOU for that link to https://artifacts.videolan.org because I found https://artifacts.videolan.org/vlc-android/release/ and I could probably point Obtainium at that.

**Comment by BRBsoup** [comment]
> > I had no problem which I've noticed with my VLC or Obtainium, which I added through this link: https://artifacts.videolan.org/vlc-android/nightly-arm64-v4/
> > My Obtainium config: [Add To Obtainium: VLC Android Nightly Arm64](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22org.videolan.vlc.debug%22%2C%22url%22%3A%22https%3A%2F%2Fartifacts.videolan.org%2Fvlc-android%2Fnightly-arm64-v4%2F%22%2C%22author%22%3A%22artifacts.videolan.org%22%2C%22name%22%3A%22VLC%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Afalse%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22APKLinkHash%5C%22%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%5C%22%2C%5C%22versionDetection%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Afalse%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Atrue%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Atrue%7D%22%2C%22overrideSource%22%3Anull%7D)
> > Last VLC update from log:
> > `2024-12-05 08:54:18.447: info: Installing "org.videolan.vlc.debug" version "4.0.0-preview - 3050740" versionCode "13050740" (from existing version "4.0.0-preview - 3050740" versionCode "13050740")`
> > ^ That was while still on Obtainium version 1.1.32 though, and there have been no new APK's added since, so...
> 
> THANK YOU for that link to https://artifacts.videolan.org because I found https://artifacts.videolan.org/vlc-android/release/ and I could probably point Obtainium at that.



> > I had no problem which I've noticed with my VLC or Obtainium, which I added through this link: https://artifacts.videolan.org/vlc-android/nightly-arm64-v4/
> > My Obtainium config: [Add To Obtainium: VLC Android Nightly Arm64](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22org.videolan.vlc.debug%22%2C%22url%22%3A%22https%3A%2F%2Fartifacts.videolan.org%2Fvlc-android%2Fnightly-arm64-v4%2F%22%2C%22author%22%3A%22artifacts.videolan.org%22%2C%22name%22%3A%22VLC%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Afalse%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22APKLinkHash%5C%22%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%5C%22%2C%5C%22versionDetection%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Afalse%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Atrue%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Atrue%7D%22%2C%22overrideSource%22%3Anull%7D)
> > Last VLC update from log:
> > `2024-12-05 08:54:18.447: info: Installing "org.videolan.vlc.debug" version "4.0.0-preview - 3050740" versionCode "13050740" (from existing version "4.0.0-preview - 3050740" versionCode "13050740")`
> > ^ That was while still on Obtainium version 1.1.32 though, and there have been no new APK's added since, so...
> 
> THANK YOU for that link to https://artifacts.videolan.org because I found https://artifacts.videolan.org/vlc-android/release/ and I could probably point Obtainium at that.

Cool :) 

https://artifacts.videolan.org/vlc-android/ has all the flavors!

I use the Nightly builds of the VLC Android v4 Alpha for Arm64-v8a cpu.

The /release page will need at least some configuration in Obtainium like adding your cpu architecture to the APK regex filter (Though if doing that then probably better off adding from Imran's config page which has all versioning done correctly - https://apps.obtainium.imranr.dev/?categories=video_player search VLC and choose cpu architecture to add to Obtainium)

If you don’t mind using untested releases then check out https://artifacts.videolan.org/vlc-android/ and find your preferred flavor by VLC version and your phone's cpu arch - Then even just add link to Obtainium with no configuration at all should work fine.

**Comment by ceskyDJ** [comment]
I use this configuration for stable releases: https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22org.videolan.vlc%22%2C%22url%22%3A%22https%3A%2F%2Fartifacts.videolan.org%2Fvlc-android%2Frelease%2F%22%2C%22author%22%3A%22artifacts.videolan.org%22%2C%22name%22%3A%22VLC%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22VLC-Android-(%5B0-9.%5D%2B)-(%3F!Beta)%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Afalse%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22APKLinkHash%5C%22%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22VLC-Android-(%5B0-9.%5D%2B)-%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%241%5C%22%2C%5C%22versionDetection%5C%22%3Atrue%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Atrue%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Afalse%2C%5C%22supportFixedAPKURL%5C%22%3Afalse%7D%22%2C%22overrideSource%22%3Anull%7D

**Comment by DarkCrypt** [comment]
https://artifacts.videolan.org/vlc-android/nightly-arm64/
The Nightly builds install.

https://artifacts.videolan.org/vlc-android/release/
The Stable builds don't install because of "invalid package".

**Comment by NicholasFlamy** [comment]
> The Stable builds don't install because of "invalid package".

Have you tried uninstalling VLC before trying to install stable?

**Comment by DarkCrypt** [comment]
> Have you tried uninstalling VLC before trying to install stable?

Yep, I tried that as well.

I looked deeper into this with App Manager and ApkTool M. I downloaded 8 different versions (3.5.4 - 3.5.7 arm64-v8a and armeabi-v7a). Both of these tools told me all 8 apks are not signed which will cause the "invalid package" install error. Same as an incomplete file will.

If there were different signatures or a signature mismatch when trying to install over a previous version, the error should say "package conflict" instead.
